apps/dashboard/app/lib/smart_attributes/attributes/auto_environment_variable.rb

```diff
@@ -0,0 +1,42 @@
+module SmartAttributes
+    class AttributeFactory
+      extend AccountCache
+  
+      def self.build_auto_environment_variable(opts = {})
+        Attributes::AutoEnvironmentVariable.new('auto_environment_variable', opts)
+      end
+    end
+  
+    module Attributes
+      class AutoEnvironmentVariable < Attribute
+        def initialize(id, opts = {})
+          super
+  
+          @key = @opts.delete(:key)
+          # reset the id to be unique from other auto_environment_variable_*
+          @id = @key ? "#{id}_#{normalize_key(@key)}" : id
+        end
+
+        def widget
+          'text_field'
+        end
+
+        def original_label
+          return 'Environment Variable' unless @opts[:label]
+          @opts[:label].to_s.gsub(": #{@key}", "")
+        end
+
+        def label(*)
+          "#{original_label}: #{@key}"
+        end
```

Comment:
I don't think we want to re-write the label if they provide it. If they provide it, we should just use it at face value because they took the time & effort to specify it.